### ISSUE TITLE
Rule proposal: No computed object property existence check

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1096]
If property name is a variable, this may leads to unexpected result.

## Fail

```js
if (foo[bar]) {}
```

In this case if `bar` is 'toString', `constructor` etc, it will be true.

## Pass

```js
if (foo[bar] === true) {}
```

```js
if (Object.prototype.hasOwnProperty.call(foo, bar)) {}
```

Problem, foo maybe an `array`, static check can't know.

Idea from https://github.com/prettier/prettier/pull/10256

**Comment by sindresorhus** [comment]
Rule name: `no-computed-property-existence-check`?

**Comment by sindresorhus** [comment]
This is now accepted.
